src/ProjectTracker1L.java

```diff
@@ -0,0 +1,110 @@
+package components.projecttracker;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Kernel implementation for ProjectTracker.
+ */
+public class ProjectTracker1L extends ProjectTrackerSecondary {
+
+    /**
+     * Representation of a milestone.
+     */
+    private static class Milestone {
+        String name;
+        String description;
+        String deadline;
+        String status;
+
+        /**
+         * Constructs a {@code Milestone} with the specified name, description,
+         * and deadline. The initial status is set to "Not started".
+         *
+         * @param name
+         * @param description
+         * @param deadline
+         */
+        Milestone(String name, String description, String deadline) {
+            this.name = name;
+            this.description = description;
+            this.deadline = deadline;
+            this.status = "Not started";
+        }
+    }
+
+    /**
+     * Representation: A map from milestone name to Milestone objects.
+     *
+     * Convention: All keys (milestone names) are unique, non-null, non-empty
+     * strings.
+     *
+     * Correspondence: Each entry in the map represents a milestone in the
+     * project tracker.
+     */
+    private Map<String, Milestone> milestones;
+
+    /**
+     * Constructor.
+     */
+    public ProjectTracker1L() {
+        this.milestones = new HashMap<>();
+    }
+
+    @Override
+    public void addMilestone(String milestoneName, String description,
+            String deadline) {
+        if (milestoneName == null || description == null || deadline == null
```

Comment:
I think doing this is fine, but it's a different approach to our discipline. You could use asserts instead and make these preconditions.